rpc/handlers_test.go

```diff
@@ -2112,3 +2113,64 @@ func TestCall(t *testing.T) {
 		assert.Equal(t, rpc.ErrContractNotFound, rpcErr)
 	})
 }
+
+func TestEstimateMessageFee(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	t.Cleanup(mockCtrl.Finish)
+
+	mockReader := mocks.NewMockReader(mockCtrl)
+	mockVM := mocks.NewMockVM(mockCtrl)
+	log := utils.NewNopZapLogger()
+
+	handler := rpc.New(mockReader, nil, utils.MAINNET, nil, nil, mockVM, "", log)
+	msg := rpc.MsgFromL1{
+		From:     common.HexToAddress("0xDEADBEEF"),
+		To:       *new(felt.Felt).SetUint64(1337),
+		Payload:  []felt.Felt{*new(felt.Felt).SetUint64(1), *new(felt.Felt).SetUint64(2)},
+		Selector: *new(felt.Felt).SetUint64(44),
+	}
+
+	t.Run("block not found", func(t *testing.T) {
+		mockReader.EXPECT().HeadState().Return(nil, nil, errors.New("not found"))
+		_, err := handler.EstimateMessageFee(msg, rpc.BlockID{Latest: true})
+		require.Equal(t, rpc.ErrBlockNotFound, err)
+	})
+
+	latestHeader := &core.Header{
+		Number:    123,
+		Timestamp: 456,
+		GasPrice:  new(felt.Felt).SetUint64(42),
+	}
+	mockState := mocks.NewMockStateHistoryReader(mockCtrl)
+	NoopCloser := func() error { return nil }
+
+	mockReader.EXPECT().HeadState().Return(mockState, NoopCloser, nil)
+	mockReader.EXPECT().HeadsHeader().Return(latestHeader, nil)
+
+	expectedGasConsumed := new(felt.Felt).SetUint64(37)
```

Comment:
How did you calculate this?

Reply:
Same question for the other values. I hadn't figured out how to test if the values themselves are correct, so if you did that would be amazing!

Reply:
Nope that is a random value. We are using a mock vm in that test, so no real execution is going on.